metanorm/normalizers/url.py

```diff
@@ -205,124 +249,59 @@ def get_instrument(self, raw_attributes):
-                    file_name_splitted[0] + '_' +
-                file_name_splitted[1] if file_name_splitted else None,
-                "ftp://nrt.cmems-du.eu/Core/GLOBAL_ANALYSIS_FORECAST_PHY_001_024/global-analysis-forecast-phy-001-024-hourly-t-u-v-ssh/":
-                    file_name_splitted[0] + '_' +
-                    file_name_splitted[-2] if file_name_splitted else None,
-                "ftp://nrt.cmems-du.eu/Core/GLOBAL_ANALYSIS_FORECAST_PHY_001_024/global-analysis-forecast-phy-001-024-monthly/":
-                    file_name_splitted[0] + '_' +
-                    file_name_splitted[-1] if file_name_splitted else None
-            }
-            extracted_date = self.extract_time(self.find_matching_value(
-                url_time, raw_attributes))
-            if not extracted_date:
-                return None
-
-            ########################################################################################
-            # further modification of "extracted time" based on other semantics of parts of the path
-            if raw_attributes['url'].startswith('ftp://ftp.remss.com'):
-                # 'd3d' are the "average of 3 days ending on and including file date"
-                # See http://www.remss.com/missions/gmi/
-                if 'd3d' in file_name:
-                    relative_days = -2 if start else 1
-                    extracted_date += relativedelta(days=relative_days)
-                # for weekly average ones in the "weeks" folder
-                # if condition is the search of "weeks" folder in the FTP path
-                elif "weeks" in url_path_and_file_name_splitted:
-                    relative_days = -6 if start else 1
-                    extracted_date += relativedelta(days=relative_days)
-                elif re.search(r"^f35_[0-9]{6}v[0-9]\.[0-9]\.gz$", file_name):
-                    # file is a month file. So, the end time must be the end of month.
-                    # python "strptime" always gives the first day. So the length of month in that
-                    # year is added.
-                    if not start:
-                        extracted_date += self.length_of_month(extracted_date)
-
-            elif raw_attributes['url'].startswith(
-                    'ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/data/CDR_v2/Climatology/L4/v2.1'):
-                # the constant date is corrected using the day
-                # number in the beginning of the file name
-                extracted_date += relativedelta(days=int(file_name[1:4]) - 1)
+            time_finder = self.find_matching_value(self.urls_time, raw_attributes)
+            if time_finder:
+                for matcher, get_time, get_coverage in time_finder:
+                    match = matcher.search(raw_attributes['url'])
+                    if match:
+                        time_info = match.groupdict()
+                        file_time = get_time(**time_info)
```

Comment:
These two lines can be merged into:
```suggestion
                        file_time = get_time(**match.groupdict())
```

Reply:
Done